File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
                HashMap <Character, Integer> cnt = new HashMap <Character, Integer>();
        int n = s.length();
        
        for (int i = 0; i < n; i++) {
            char c = s.charAt(i);
            cnt.put(c, cnt.getOrDefault(c, 0) + 1);
        }
        
        for (int i = 0; i < n; i++){
            if (cnt.get(s.charAt(i)) == 1)
                return i;
        }
        return -1;
    }
}

File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        
        unordered_map <char,int> alphabets(26);
        
        for (int i = 0; i < s.size(); i++){
            alphabets[s[i]]++;
        }
        
        for (int i = 0; i < s.size(); i++){
            if (alphabets[s[i]] == 1) 
                return i;
        }
        return -1;
    }
};